tarantool_test.go

```diff
@@ -2397,6 +2447,112 @@ func TestClientRequestObjects(t *testing.T) {
 	}
 }
 
+func testConnectionDoSelectRequestPrepare(t *testing.T, conn Connector) {
+	t.Helper()
+
+	for i := 1010; i < 1020; i++ {
+		deleteReq := NewDeleteRequest(spaceName).Key([]interface{}{uint(i)})
+		insertReq := NewInsertRequest(spaceName).Tuple(
+			[]interface{}{uint(i), fmt.Sprintf("val %d", i), "bla"})
+		conn.Do(deleteReq).Get()
+		if _, err := conn.Do(insertReq).Get(); err != nil {
+			t.Fatalf("Unable to prepare tuples: %s", err)
+		}
+	}
+}
+
+func TestConnectionDoSelectRequest(t *testing.T) {
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	testConnectionDoSelectRequestPrepare(t, conn)
+
+	req := NewSelectRequest(spaceNo).
+		Index(indexNo).
+		Limit(20).
+		Iterator(IterGe).
+		Key([]interface{}{uint(1010)})
+
+	resp, err := conn.Do(req).Get()
+	if err != nil {
+		t.Fatalf("Failed to Select: %s", err.Error())
+	}
+	if resp == nil {
+		t.Fatalf("Response is nil after Select")
+	}
+	if resp.Pos != nil {
+		t.Errorf("Response should not have a position")
+	}
+	if len(resp.Data) != 10 {
+		t.Fatalf("Response Data len %d != 9", len(resp.Data))
+	}
+	if tpl, ok := resp.Data[0].([]interface{}); !ok {
+		t.Errorf("Unexpected body of Select")
+	} else {
+		if id, err := convertUint64(tpl[0]); err != nil || id != 1010 {
+			t.Errorf("Unexpected body of Select (0) %v, expected %d", tpl[0], 1010)
+		}
+		if h, ok := tpl[1].(string); !ok || h != "val 1010" {
+			t.Errorf("Unexpected body of Select (1) %q, expected %q", tpl[1].(string), "val 1010")
+		}
+		if h, ok := tpl[2].(string); !ok || h != "bla" {
+			t.Errorf("Unexpected body of Select (2) %q, expected %q", tpl[2].(string), "bla")
+		}
+	}
+}
+
+func TestConnectionDoSelectRequest_pagination(t *testing.T) {
+	test_helpers.SkipIfPaginationUnsupported(t)
+
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	testConnectionDoSelectRequestPrepare(t, conn)
+
+	req := NewSelectRequest(spaceNo).
+		Index(indexNo).
+		Limit(2).
+		Iterator(IterGe).
+		FetchPos(true).
+		Key([]interface{}{uint(1010)})
+
+	resp, err := conn.Do(req).Get()
+	if err != nil {
+		t.Fatalf("Failed to Select: %s", err.Error())
+	}
+	if resp.Pos == nil {
+		t.Fatalf("A response must have a position descriptor")
+	}
+	if len(resp.Data) != 2 {
+		t.Fatalf("Response Data len %d != 2", len(resp.Data))
+	}
+
+	for i := 0; i < 2; i++ {
```

Comment:
To be honest, this code is confusing. If you want to evade copypaste, maybe extract it to a separate function?

Reply:
I split the test into several.